2001-03-16  Jakub Jelinek  <jakub@redhat.com>

	* reg-stack.c (check_asm_stack_operands): Issue error if "=t" resp.
	"=u" is used together with "st" resp. "st(1)" clobber.

--- gcc/reg-stack.c.jj	Thu Mar 15 13:03:12 2001
+++ gcc/reg-stack.c	Fri Mar 16 14:01:29 2001
@@ -658,7 +658,20 @@ check_asm_stack_operands (insn)
 	    malformed_asm = 1;
 	  }
         else
-	  reg_used_as_output[REGNO (recog_data.operand[i])] = 1;
+	  {
+	    int j;
+
+	    for (j = 0; j < n_clobbers; j++)
+	      if (REGNO (recog_data.operand[i]) == REGNO (clobber_reg[j]))
+		{
+		  error_for_asm (insn, "Output constraint %d cannot be specified together with \"%s\" clobber",
+				 i, reg_names [REGNO (clobber_reg[j])]);
+		  malformed_asm = 1;
+		  break;
+		}
+	    if (j == n_clobbers)
+	      reg_used_as_output[REGNO (recog_data.operand[i])] = 1;
+	  }
       }
 
 
